### PR TITLE
[8.2] make AddAllowDynamicPropertiesAttributeRector configurable

### PR DESCRIPTION
**Pull request by mallardduck**
Per the talk in #1225 this expands on that by making it configurable for namespaces.

### CODE REVIEW

File: rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php [hunk 1]
```diff
@@ -7,5 +7,12 @@
 
 return static function (ContainerConfigurator $containerConfigurator): void {
     $services = $containerConfigurator->services();
-    $services->set(AddAllowDynamicPropertiesAttributeRector::class);
+    $services->set(AddAllowDynamicPropertiesAttributeRector::class)
+        ->call('configure', [[
+            [
```
Comment (mallardduck) [inline rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php:12]: @samsonasik - I know the docs say it should start with `[[` however for some reason when I do that the `configure` method gets the wrong value. By adding a third level of array it's working as expected.
Any thoughts on that?

---

Without the "extra" array I get:
```
array(1) {
  [0] =>
  string(19) "*\Fixture\Process\*"
}
```

With the extra array:
```
array(1) {
  'transform_on_namespaces' =>
  array(1) {
    [0] =>
    string(19) "*\Fixture\Process\*"
  }
}
```

Comment (mallardduck) [reply rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php:12]: Actually, never mind - I see now that there's some sort of magic going on there. Where when the array contains a single member whose value is an array it unwraps to that member.

Comment (mallardduck) [reply rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php:12]: NVM. Got that sorted out by looking at existing rules.

File: rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php [hunk 2]
```diff
@@ -7,5 +7,10 @@
 
 return static function (ContainerConfigurator $containerConfigurator): void {
     $services = $containerConfigurator->services();
-    $services->set(AddAllowDynamicPropertiesAttributeRector::class);
+    $services->set(AddAllowDynamicPropertiesAttributeRector::class)
```
Comment (samsonasik) [inline rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php:10]: please add 1 more test class with no config to fallback to change all classes, even without namespace

File: rules/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector.php [hunk 3]
```diff
@@ -8,28 +8,37 @@
...
 use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
 
 /**
  * @changelog https://wiki.php.net/rfc/deprecate_dynamic_properties
  *
  * @see \Rector\Tests\Transform\Rector\Class_\AddAllowDynamicPropertiesAttributeRector\AddAllowDynamicPropertiesAttributeRectorTest
  */
-final class AddAllowDynamicPropertiesAttributeRector extends AbstractRector implements MinPhpVersionInterface
+final class AddAllowDynamicPropertiesAttributeRector extends AbstractRector implements ConfigurableRectorInterface, MinPhpVersionInterface
```
Comment (samsonasik) [inline rules/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector.php]: please rebase and change to implements `\Rector\Core\Contract\Rector\AllowEmptyConfigurableRectorInterface` so it won't show warning when not configured.

File: rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php [hunk 4]
```diff
@@ -7,5 +7,10 @@
...
+        ->call('configure', [[
+            AddAllowDynamicPropertiesAttributeRector::TRANSFORM_ON_NAMESPACES => [
+                '*\Fixture\Process\*'
+            ]
+        ]]);
```
Comment (samsonasik) [inline rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php]: can be changed to:

```suggestion
        ->configure(
            [
                '*\Fixture\Process\*'
            ]
        );
```

Comment (TomasVotruba) [reply rules-tests/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector/config/configured_rule.php]: We should have probably PHPStan rule for this one :+1: 

In https://github.com/rectorphp/phpstan-rules

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thanks again :clap: :+1:

**Pull request merged**